File: orchard.java
import java.util.Scanner;

public class orchard {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Input trees for Ashok and Anand
        System.out.print("Enter Ashok's row of trees: ");
        String ashokTrees = scanner.nextLine().toUpperCase();

        System.out.print("Enter Anand's row of trees: ");
        String anandTrees = scanner.nextLine().toUpperCase();

        // Validate input strings
        if (!isValidInput(ashokTrees) || !isValidInput(anandTrees)) {
            System.out.println("Invalid input");
            return;
        }

        int ashokPossibilities = countPossibilities(ashokTrees);
        int anandPossibilities = countPossibilities(anandTrees);

        if (ashokPossibilities > anandPossibilities) {
            System.out.println("Ashok");
        } else if (anandPossibilities > ashokPossibilities) {
            System.out.println("Anand");
        } else {
            System.out.println("Draw");
        }
    }

    static boolean isValidInput(String trees) {
        return trees.matches("[ML]+");
    }

    static int countPossibilities(String trees) {
        int possibilities = 0;
        for (int i = 0; i < trees.length() - 2; i++) {
            if (trees.charAt(i) != trees.charAt(i + 1) && trees.charAt(i + 1) != trees.charAt(i + 2)
                    && trees.charAt(i) != trees.charAt(i + 2)) {
                possibilities++;
            }
        }

        return possibilities;
    }
}

File: orchard.cpp
#include <iostream>
#include <cstring>

using namespace std;

int combination(char arr[], int n) {
    int orc = 0;
    string s = "";
    for (int i = 0; i < n; i++) {
        s = arr[i];
        for (int j = i + 1; j < n; j++) {
            s += arr[j];
            if (s[0] != arr[j]) {
                for (int k = j + 1; k < n; k++) {
                    if (s[1] != arr[k])
                        orc++;
                }
            }
            s = arr[i];
        }
    }
    return orc;
}

int main() {
    string ashok, anand;
    getline(cin, ashok);
    getline(cin, anand);

    char as[ashok.length()];
    char an[anand.length()];

    strcpy(as, ashok.c_str());
    strcpy(an, anand.c_str());

    int orc_as = combination(as, ashok.length());
    int orc_an = combination(an, anand.length());

    if (orc_as == 0 && orc_an == 0)
        cout << "Invalid input";
    else if (orc_as > orc_an)
        cout << "Ashok";
    else if (orc_an > orc_as)
        cout << "Anand";
    else
        cout << "Draw";

    return 0;
}
